scripts/workflow-parameter-extraction.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+import argparse
+
+import WDL
+import os
+import sys
+
+input_wdl_path = None
+output_config_path = None
+
+doc = None  # WDL document object
+
+
+def trace_struct_declarations(obj, struct_type_info={}):
+    if isinstance(obj, WDL.Document):
+        if obj.workflow:
+            trace_struct_declarations(obj.workflow, struct_type_info)
+        for task in obj.tasks:
+            trace_struct_declarations(task, struct_type_info)
+    elif isinstance(obj, WDL.Workflow):
+        for child in (obj.inputs or []) + obj.body:
+            trace_struct_declarations(child)
+    elif isinstance(obj, WDL.Task):
+        for child in (
+            (obj.inputs or [])
+            + obj.postinputs
+            + [obj.command]
+            + list(obj.runtime.values())
+        ):
+            trace_struct_declarations(child)
+    elif isinstance(obj, WDL.WorkflowSection):
+        trace_struct_declarations(obj.expr, struct_type_info)
+        for child in obj.body:
+            trace_struct_declarations(child, struct_type_info)
+    elif isinstance(obj, WDL.Call):
+        for rhs in obj.inputs.values():
+            trace_struct_declarations(rhs, struct_type_info)
+    elif isinstance(obj, WDL.Decl):
+        trace_struct_declarations(obj.expr, struct_type_info)
+    elif isinstance(obj, WDL.Expr.Base):
+        for child in obj.children:
+            trace_struct_declarations(child, struct_type_info)
+
+    if isinstance(obj, WDL.Expr.Ident) and isinstance(obj.type, WDL.Type.Array):
+        var_name = str(doc.workflow.name) + "." + str(obj.name)
+        for param in obj.type.parameters:
+            if isinstance(param, WDL.Type.StructInstance):
+                struct_fields = []
```

Comment:
I think these two struct fields are supposed to be a dictionary of member name to member type rather than a list.